ddmrp/models/stock_buffer.py

```diff
@@ -1194,23 +1196,27 @@ def write(self, vals):
     @api.model
     def cron_ddmrp_adu(self, automatic=False):
         """calculate ADU for each DDMRP buffer. Called by cronjob."""
+        auto_commit = not getattr(threading.currentThread(), "testing", False)
         _logger.info("Start cron_ddmrp_adu.")
-        buffers = self.search([])
+        buffer_ids = self.search([]).ids
         i = 0
-        j = len(buffers)
-        for b in buffers:
-            try:
-                i += 1
-                _logger.debug("ddmrp cron_adu: {}. ({}/{})".format(b.name, i, j))
-                if automatic:
-                    with self.env.cr.savepoint():
+        j = len(buffer_ids)
+        for buffer_chunk_ids in split_every(50, buffer_ids):
+            for b in self.browse(buffer_chunk_ids):
```

Comment:
```suggestion
            for b in self.browse(buffer_chunk_ids).exists():
```
In case a buffer has been deleted.